src/core/utils/channel_monitor.cpp

```diff
@@ -213,6 +213,55 @@ void ChannelMonitor::LogResults(void)
         mChannelQuality[15] >> 8);
 }
 
+Mac::ChannelMask ChannelMonitor::FindBestChannels(const Mac::ChannelMask &aMask, uint16_t &aBestQuality)
+{
+    uint8_t          channel;
+    Mac::ChannelMask bestMask;
+
+    bestMask.Clear();
+    aBestQuality = 0xffff;
+
+    channel = Mac::ChannelMask::kChannelIteratorFirst;
+
+    while (aMask.GetNextChannel(channel) == OT_ERROR_NONE)
+    {
+        uint16_t quality = GetChannelQuality(channel);
+
+        if (bestMask.IsEmpty() || (quality <= aBestQuality))
+        {
+            if (quality < aBestQuality)
```

Comment:
It's a bit confusing to have lower integer values actually have higher quality.  I would consider renaming or changing the polarity.